extensions/common/vault/vault-aws/src/test/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultExtensionTest.java

```diff
@@ -16,35 +16,58 @@
 
 import org.eclipse.edc.spi.monitor.Monitor;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
-import org.eclipse.edc.spi.system.configuration.Config;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
 
+import java.net.URI;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.type;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 class AwsSecretsManagerVaultExtensionTest {
+    private static ServiceExtensionContext context;
+
+    @BeforeAll
+    public static void beforeAll() {
+        context = mock(ServiceExtensionContext.class);
+        when(context.getMonitor()).thenReturn(mock(Monitor.class));
+    }
 
-    private final Monitor monitor = mock(Monitor.class);
-    private final AwsSecretsManagerVaultExtension extension = new AwsSecretsManagerVaultExtension();
 
     @Test
     void configOptionRegionNotProvided_shouldThrowException() {
-        ServiceExtensionContext invalidContext = mock(ServiceExtensionContext.class);
-        when(invalidContext.getMonitor()).thenReturn(monitor);
+        var extension = new AwsSecretsManagerVaultExtension();
 
-        Assertions.assertThrows(NullPointerException.class, () -> extension.createVault(invalidContext));
+        Assertions.assertThrows(NullPointerException.class, () -> extension.createVault(context));
     }
 
     @Test
     void configOptionRegionProvided_shouldNotThrowException() {
-        ServiceExtensionContext validContext = mock(ServiceExtensionContext.class);
-        Config cfg = mock();
-        when(cfg.getString("edc.vault.aws.region")).thenReturn("eu-west-1");
-        when(validContext.getConfig()).thenReturn(cfg);
-        when(validContext.getMonitor()).thenReturn(monitor);
+        var extension = new AwsSecretsManagerVaultExtension();
+        extension.vaultRegion = "eu-west-1";
+
+        var vault = extension.createVault(context);
 
-        extension.createVault(validContext);
+        assertThat(vault).extracting("smClient", type(SecretsManagerClient.class)).satisfies(client -> {
+            assertThat(client.serviceClientConfiguration().region()).isEqualTo(Region.of("eu-west-1"));
+        });
     }
 
+    @Test
+    void configOptionEndpointOverrideProvided_shouldNotThrowException() {
+        var extension = new AwsSecretsManagerVaultExtension();
+        extension.vaultRegion = "eu-west-1";
+        extension.vaultAwsEndpointOverride = "http://localhost:4566";
```

Comment:
I'm trying to use `key` of `@Setting`, but I can't find the way to inject the testing value to the field variable. Please let me know how to test `@Setting` annotated value with `key` overrides if you know a better way!

Reply:
My suggestion is to use the `DependencyInjectionExtension` that can be found in the `edc-junit` module.

Then this test can be written this way:
```java
    @Test
    void configOptionEndpointOverrideProvided_shouldNotThrowException(ObjectFactory factory, ServiceExtensionContext context) {
        var config = ConfigFactory.fromMap(Map.of(
                "edc.vault.aws.region", "region",
                "edc.vault.aws.endpoint.override", "http://localhost:4566"
        ));
        when(context.getConfig()).thenReturn(config);
        var extension = factory.constructInstance(AwsSecretsManagerVaultExtension.class);

        var vault = extension.createVault(context);

        assertThat(vault).extracting("smClient", type(SecretsManagerClient.class)).satisfies(client -> {
            assertThat(client.serviceClientConfiguration().endpointOverride()).contains(URI.create("http://localhost:4566"));
        });
    }
```

and no need to make the setting fields package private